components/_Progress-Bar/progress-bar.js

```diff
@@ -0,0 +1,63 @@
+const msdsProgressBar = {
+  //Get the base number for the progress bar
+  parentElement: '',
+  baseNumber: 0,
+  currentStep: 0,
+  startFromOne: false,
+  init: function(parentElementId, baseNumber, customProgressionStep) {
+    this.parentElement = document.getElementById(parentElementId)
+
+    if (customProgressionStep > 0) {
+      this.currentStep = customProgressionStep
+    } else if (this.startFromOne == true) {
+      this.currentStep = 1
+    } else {
+      this.currentStep = 0
+    }
+    this.baseNumber = baseNumber
+    this.updateProgressbarDisplay()
+  },
+  updateProgressbarDisplay: function() { //update the display of the prorgess bar (step and counter)
+    
+    const progressBarProgressionElement = this.parentElement.querySelector('.msds-progress-bar__progression-bar')
+    if (progressBarProgressionElement) {
+      //calculate the percentage width of the progress step and update the look and fill base on this
+      const updatedWidth = (this.currentStep * 100) / this.baseNumber
+      progressBarProgressionElement.style.width = updatedWidth + '%'
```

Comment:
Personally prefer string interpolation in this way:
```
progressBarProgressionElement.style.width = `${updatedWidth}%`
```

and + operator use only for numbers. But is nothing wrong with this, you can try if you want.